readme.md

```diff
@@ -1,38 +1,49 @@
 # filter mod
 
-This mod adds a simple chat filter. There is no default word list,
-and adding words to the filter list is done through the `/filter`
-chat command. You need the `server` priv to use the chat command.
+This mod adds a chat filter API that can be used to register custom
+chat filters. As of now a bad words filter has been implemented. There
+is no default word list, and adding words to the filter list is done
+through the `/word_filter` chat command, which requires `server` priv.
 
-The `/filter` chat command can `add`, `remove` or `list` words. The
+The `/word_filter` chat command can `add`, `remove` or `list` words. The
 words are stored in `mod_storage`, which means that this mod requires
 0.4.16 or above to function.
 
-If a player speaks a word that is listed in the filter list, they are
-muted for 1 minute. After that, their `shout` privilege is restored.
-If they leave, their `shout` privilege is still restored, but only after
-the time expires, not before.
+If a player triggers a filter, they are muted for 1 minute. After that,
+their `shout` privilege is restored. If they leave, their `shout`
+privilege is still restored, but only after the time expires, not before.
 
 ## API
 
+### Custom filter registration
+
+* `filter.register_filter(fname, func(pname, message))`
+  * Takes a name and a two-parameter function
+  * `fname` is the name of the filter, which is currently unused, except
+    for indexing.
+  * `pname` and `message` is the name of the player and the message they
+    sent.
+  * `func` should return a relevant warning when triggered. e.g.
+    "Watch your language!".
+
 ### Callbacks
 
-* filter.register_on_violation(func(name, message, violations))
-	* Violations is the value of the player's violation counter - which is
-	  incremented on a violation, and halved every 10 minutes.
-	* Return true if you've handled the violation. No more callbacks will be
-	  executation, and the default behaviour (warning/mute/kick) on violation
-	  will be skipped.
+* `filter.register_on_violation(func(name, message, violations))`
+  * `violations` is the value of the player's violation counter - which is
+    incremented on a violation, and halved every 10 minutes.
```

Comment:
this needs a setting, and I'm not entirely sure I like this at all.

Reply:
I think this should be a separate PR. There are many more variables that can be made configurable.